v2/googlecloud-to-elasticsearch/docs/GCSToElasticsearch/README.md

```diff
@@ -11,11 +12,35 @@ If either a UDF or JSON schema is provided then it will be used instead of the C
 
 \*<b>NOTE:</b> All values will be written to Elasticsearch as strings if headers are used.
 
-Pipeline flow is illustrated below:
+## Getting Started
 
-![alt text](../img/gcs-to-elasticsearch-dataflow.png "GCS to Elasticsearch pipeline flow")
+The simplest way to set up Elasticsearch is to create a managed deployment with Elasticsearch Service on [Elastic Cloud](https://www.elastic.co/guide/en/elasticsearch/reference/current/getting-started.html#run-elasticsearch). You can deploy Elasticsearch on many [Google Cloud regions](https://www.elastic.co/guide/en/cloud/current/ec-regions-templates-instances.html) globally. If you prefer to manage your own environment, you can install and run [Elasticsearch using Docker](https://www.elastic.co/guide/en/elasticsearch/reference/current/getting-started.html#run-elasticsearch).
 
-## Getting Started
+### Parameters
+
+The template requires the following parameters:
+* connectionUrl: Elasticsearch URL in format http://hostname:[port] or CloudId
+* inputFileSpec: Pattern to where data lives in GCS. Example: `gs://mybucket/somepath/*.csv`
+* errorOutputTopic: Error output topic in Pub/Sub for failed inserts
+* apiKey: Base64 Encoded API Key for access without requiring basic authentication. Refer  https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-create-api-key.html#security-api-create-api-key-request.
+* index: Destination index in Elasticsearch
+* deadletterTable: Deadletter table in BigQuery for failed inserts in form: project-id:dataset.table
+
+The template has the following optional parameters:
+
+* containsHeaders: Set to true if file(s) contain headers.
+* delimiter: Delimiting character in CSV file(s). Default: use delimiter found in csvFormat
+* inputFormat: The format of the input files, default is CSV.
+* csvFormat: Csv format according to [Apache Commons CSV format](https://commons.apache.org/proper/commons-csv/apidocs/org/apache/commons/csv/CSVFormat.html). Default is: Default
+  <br><b>NOTE:</b> Must match format names exactly found [here](http://commons.apache.org/proper/commons-csv/apidocs/org/apache/commons/csv/CSVFormat.Predefined.html#Default)
+* jsonSchemaPath: Path to JSON schema, ex gs://path/to/schema. Default: null.
+* largeNumFiles: Set to true if number of files is in the tens of thousands. Default: false
+* batchSize: Batch size in number of documents. Default: 1000
+* batchSizeBytes: Batch size in number of bytes. Default: 5242880 (5mb)
+* javascriptTextTransformGcsPath: Gcs path to javascript udf source. Udf will be preferred option for transformation if supplied. Default: null
+* javascriptTextTransformFunctionName: UDF Javascript Function Name. Default: null
+* maxRetryAttempts: Max retry attempts, must be > 0. Default: no retries
+* maxRetryDuration: Max retry duration in milliseconds, must be > 0. Default: no retries
 
 ### Requirements
```

Comment:
Instead of showcasing ELASTICSEARCH_USERNAME/ELASTICSEARCH_PASSWORD, ideally apiKey should be projected as an authentication mechanism.  Better to remove below statements
```
export ELASTICSEARCH_USERNAME=<username>
export ELASTICSEARCH_PASSWORD=<password>
```
and add 
```
export API_KEY=<Base64 encoded api key>
```